pilot/pkg/security/authz/model/principal.go

```diff
@@ -59,9 +59,12 @@ func principalNot(principal *rbacpb.Principal) *rbacpb.Principal {
 
 func principalAuthenticated(name *matcher.StringMatcher) *rbacpb.Principal {
 	return &rbacpb.Principal{
-		Identifier: &rbacpb.Principal_Authenticated_{
-			Authenticated: &rbacpb.Principal_Authenticated{
-				PrincipalName: name,
+		Identifier: &rbacpb.Principal_FilterState{
```

Comment:
Will replacing Principal_Authenticated_ with Principal_FilterState impact any authn&authz workflows?

Reply:
Good question. Not in the supported cases AFAICT. This is what "Authenticated" means: 
` If set, The URI SAN or DNS SAN in that order is used from the certificate, otherwise the subject field is used`.

DNS SAN is never used because Istio programs RBAC to match URI patterns (SPIFFE). So the main difference is the case of multiple SPIFFE certs in URI SAN list. The filter state will choose the first URI SAN, while `Authenticated` will choose any. 

I couldn't find any good information on certs with multiple identities. I think many things go wrong in those cases (e.g. telemetry doesn't match RBAC).

Reply:
See SPIFFE specification:
```
 An X.509 SVID MUST contain exactly one URI SAN, and by extension, exactly one SPIFFE ID. SVIDs containing more than one SPIFFE ID introduce challenges related to auditing and authorization logic, and SVIDs containing more than one URI SAN introduce challenges related to SPIFFE ID validation. Validators encountering an SVID containing more than one URI SAN MUST reject the SVID.
```
(Ref: https://github.com/spiffe/spiffe/blob/main/standards/X509-SVID.md#2-spiffe-id)

So it's recommended to use only one SPIFFE ID from URI SANs and ignore DNS SANs. This PR actually brings us closer to the recommendation.

Reply:
Thanks for the detailed explanations!

Reply:
Where's this config? There doesn't seem to be any test for DNS SAN, and it's impossible to match DNS with Istio translator (it adds a slash).

Reply:
clientTlsSettings.subjectAltNames - list of SANs to verify ( for client side ) - there is one in ServiceEntry as well.
I believe it now checks both DNS and URL SANs - and is not restricted to spiffee.

That's for client authenticating the server.

For authz - you are right, we don't seem to support non-spiffee. Not entirely sure if source.principal example is meant to indicate only this format is supported - or if a user could say 'principal=costin@example.com' - i.e. accept the common certs with email SAN. 

Yes, tests for non spiffe are likely to be missing :-)